kit/inputs/src/useApplyMask/useApplyMask.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-param-reassign */
+import * as React from 'react';
+import { useSharedValue } from 'react-native-reanimated';
+import type { UIMaterialTextViewMask, ChangeText, MoveCarret } from '../UIMaterialTextView/types';
+import { onChangeAmount } from './amount';
+import { useOnSelectionChange } from './useOnSelectionChange';
+
+export function useApplyMask(
+    changeText: ChangeText,
+    moveCarret: MoveCarret,
+    mask?: UIMaterialTextViewMask,
+) {
+    const { selectionStart, selectionEnd, onSelectionChange, skipNextOnSelectionChange } =
+        useOnSelectionChange();
+
+    const lastNormalizedText = useSharedValue('');
+    const lastText = useSharedValue('');
+
+    const onChangeText = React.useCallback(
+        (text: string): void => {
+            switch (mask) {
+                case '[000] [000].[000] [000]':
+                    onChangeAmount(
```

Comment:
So, it's basically just a custom name now)

Reply:
This is a temporary solution so as not to complicate this task